gateway-topology-simple/src/main/java/org/apache/knox/gateway/topology/simple/SimpleDescriptorHandler.java

```diff
@@ -635,6 +635,15 @@ private static Map<String, File> generateTopology(final SimpleDescriptor desc,
         return result;
     }
 
+    /*
+     * First, undoes any previous manual XML escape.
+     * Second applies XML-escape on the result of the first step.
+     */
+    private static String getXmlEscapedValue(String value) {
+      final String unescapedValue = StringEscapeUtils.unescapeXml(value);
```

Comment:
I wonder if a "cheap" check for escaped values is worthwhile here before unescaping.

Reply:
I had this idea too, but [`StringEscapeUtils`](https://commons.apache.org/proper/commons-text/apidocs/org/apache/commons/text/StringEscapeUtils.html) does not have any test method to see if the given value requires unescape.

Reply:
Perhaps, even just a check for '&' since I don't think that's expected in unescaped forms of this content?

Reply:
Proper parsing for &XXX; is probably more expensive than the unescaping.